include/boost/geometry/strategies/cartesian/intersection.hpp

```diff
@@ -245,6 +247,20 @@ struct cartesian_segments
                                                          / denominator));
         }
 
+        template <int Index, typename Point, typename Segment>
+        static bool exceeds(Point& point, Segment const& s)
+        {
+            return (geometry::get<Index, 0>(s) < geometry::get<1 - Index, 0>(s)
+                    && geometry::get<0>(point) < geometry::get<Index, 0>(s))
+                || (geometry::get<Index, 0>(s) > geometry::get<1 - Index, 0>(s)
+                    && geometry::get<0>(point) > geometry::get<Index, 0>(s))
+                || (geometry::get<Index, 1>(s) < geometry::get<1 - Index, 1>(s)
+                    && geometry::get<1>(point) < geometry::get<Index, 1>(s))
+                || (geometry::get<Index, 1>(s) > geometry::get<1 - Index, 1>(s)
+                    && geometry::get<1>(point) > geometry::get<Index, 1>(s));
```

Comment:
I can make an extra function per dimension such that will look simpler, and add some ascii art

Reply:
:heavy_check_mark: done